modules/etcd/etcd_unit_test.go

```diff
@@ -0,0 +1,103 @@
+package etcd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/errdefs"
+	"github.com/stretchr/testify/require"
+
+	"github.com/testcontainers/testcontainers-go"
+	tcexec "github.com/testcontainers/testcontainers-go/exec"
+)
+
+func TestRunCluster1Node(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := Run(ctx, "gcr.io/etcd-development/etcd:v3.5.14")
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	// the topology has only one node with no children
+	require.Empty(t, ctr.childNodes)
+	require.Equal(t, defaultClusterToken, ctr.opts.clusterToken)
+}
+
+func TestRunClusterMultipleNodes(t *testing.T) {
+	testCases := []struct {
+		name  string
+		node1 string
+		node2 string
+		nodes []string
+	}{
+		{
+			name:  "2-nodes",
+			node1: "etcd-1",
+			node2: "etcd-2",
+			nodes: []string{},
+		},
+		{
+			name:  "3-nodes",
+			node1: "etcd-1",
+			node2: "etcd-2",
+			nodes: []string{"etcd-3"},
+		},
+	}
```

Comment:
suggestion: if you extract the main body to a function you can use inline test which is significantly cleaner and the benefit of being able to run individual cases in editors e.g.
```suggestion
	t.Run("2-nodes", testCluster(t, "etcd-1", "etcd-1", "etcd-3"))
	t.Run("3-nodes", testCluster(t, "etcd-1", "etcd-1", "etcd-3"))
```

The Run call will look something like:
```go
ctr, err := Run(ctx, "gcr.io/etcd-development/etcd:v3.5.14", WithNodes(nodes[0], nodes[1], nodes[2:]...), WithClusterToken(clusterToken))
```

Reply:
I agree that inlining is cleaner 👍 OTOH my IDE allows my to run subtests even with this test case layout. I'm using VSCode. What IDE do you use?

Reply:
> "2-nodes", testCluster(t, "etcd-1", "etcd-1", "etcd-3")

Is this on purpose a bug? declaring a two-node cluster with 3 parameters? This makes me think if we need to validate the node names are unique 🤔 Wdyt?

Reply:
Added in https://github.com/testcontainers/testcontainers-go/pull/2788/commits/f52460d030715fd67baecee875f9bcac10141e35

Reply:
VSCode for me you can only select subtests when they are explicitly named, the for loop prevents you running just the subtest for example: `TestRunClusterMultipleNodes/2-nodes` and not just the top level test `TestRunClusterMultipleNodes`.

Reply:
That example was just a typo